hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/cancelairdrops/TokenCancelAirdropDecoderTest.java

```diff
@@ -0,0 +1,327 @@
+
+@ExtendWith(MockitoExtension.class)
+class TokenCancelAirdropDecoderTest {
+
+    @Mock
+    private HtsCallAttempt attempt;
+
+    @Mock
+    private AddressIdConverter addressIdConverter;
+
+    @Mock
+    private Enhancement enhancement;
+
+    @Mock
+    private HederaNativeOperations nativeOperations;
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private TokensConfig tokensConfig;
+
+    private TokenCancelAirdropDecoder subject;
+
+    @BeforeEach
+    void setup() {
+        subject = new TokenCancelAirdropDecoder();
+
+        lenient().when(attempt.addressIdConverter()).thenReturn(addressIdConverter);
+    }
+
+    @Test
+    void cancelAirdropDecoder1FTTest() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(attempt.enhancement()).willReturn(enhancement);
+        given(enhancement.nativeOperations()).willReturn(nativeOperations);
+        given(configuration.getConfigData(TokensConfig.class)).willReturn(tokensConfig);
+        given(tokensConfig.maxAllowedPendingAirdropsToCancel()).willReturn(10);
+        given(nativeOperations.getToken(FUNGIBLE_TOKEN_ID.tokenNum())).willReturn(FUNGIBLE_TOKEN);
+        given(addressIdConverter.convert(asHeadlongAddress(SENDER_ID.accountNum())))
+                .willReturn(SENDER_ID);
+        given(addressIdConverter.convert(OWNER_ACCOUNT_AS_ADDRESS)).willReturn(OWNER_ID);
+
+        final var encoded = Bytes.wrapByteBuffer(CANCEL_AIRDROP.encodeCall(Tuple.singleton(new Tuple[] {
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L)
+        })));
+        given(attempt.inputBytes()).willReturn(encoded.toArrayUnsafe());
+
+        var expected = new ArrayList<PendingAirdropId>();
+        expected.add(PendingAirdropId.newBuilder()
+                .senderId(SENDER_ID)
+                .receiverId(OWNER_ID)
+                .fungibleTokenType(FUNGIBLE_TOKEN_ID)
+                .build());
+        // when:
+        final var decoded = subject.decodeCancelAirdrop(attempt);
+
+        // then:
+        assertNotNull(decoded.tokenCancelAirdrop());
+        assertEquals(expected, decoded.tokenCancelAirdrop().pendingAirdrops());
+    }
+
+    @Test
+    void failsIfPendingAirdropsAboveLimit() {
+        // given:
+        given(attempt.configuration()).willReturn(configuration);
+        given(configuration.getConfigData(TokensConfig.class)).willReturn(tokensConfig);
+        given(tokensConfig.maxAllowedPendingAirdropsToCancel()).willReturn(10);
+
+        final var encoded = Bytes.wrapByteBuffer(CANCEL_AIRDROP.encodeCall(Tuple.singleton(new Tuple[] {
+            Tuple.of(
+                    asHeadlongAddress(SENDER_ID.accountNum()),
+                    OWNER_ACCOUNT_AS_ADDRESS,
+                    FUNGIBLE_TOKEN_HEADLONG_ADDRESS,
+                    0L),
```

Comment:
Could, if you wanted, just put this in a local and then put that same instance into the array 11 times.  Nobody is deduping the array ....

Could also reduce the limit below 10 ...

Reply:
Done